packages/config/src/Config.ts

```diff
@@ -526,15 +532,19 @@ export class Config {
                 }
             }
 
+            // Clear the entry and rebuild it
+            delete this._secure.configs[layer.path];
+
             // Create the entry to set the secure properties
             if (Object.keys(sp).length > 0) {
                 this._secure.configs[layer.path] = sp;
             }
```

Comment:
So, this prevents `secureSave` from storing empty objects in the vault? (while at the same time clearing any empty objects already there?)

Reply:
Yes. The idea is to reduce the size of the object stored on the vault if there are no secure properties, and to delete any properties that were on the vault before if they were all removed.